### PR TITLE
Use CSS z-index to order layers

### PR DESCRIPTION
**Pull request by fredj**

### CODE REVIEW

File: src/ol/renderer/Composite.js [hunk 1]
```diff
@@ -98,6 +95,10 @@ class CompositeMapRenderer extends MapRenderer {
       const layerRenderer = this.getLayerRenderer(layer);
       if (layerRenderer.prepareFrame(frameState, layerState)) {
         const element = layerRenderer.renderFrame(frameState, layerState);
+        const zIndex = layerState.zIndex;
+        if (zIndex !== element.style.zIndex) {
```
Comment (tonio) [inline src/ol/renderer/Composite.js]: Is this check really necessary? I don't think set an identical `zIndex` will have an effect…

Comment (fredj) [reply src/ol/renderer/Composite.js]: My idea is that changing the `zIndex` even if it's the same value will trigger a reflow or something similar (but I haven't tried if it's really the case)

Comment (tonio) [reply src/ol/renderer/Composite.js]: I just test to confirm, looks like it isn't the case.

**Comment by tschaub** [comment]
@fredj it should be pretty easy to fix up these conflicts if we want to get this in.

**Comment by fredj** [comment]
@tschaub done

**Pull request merged**